lib/components/admin/editable-section.js

```diff
@@ -0,0 +1,153 @@
+import React, {Component} from 'react'
+
+import {
+  Button,
+  P,
+  Val
+} from './styled'
+
+/**
+ * This component will render a set of fields along with a 'Change/Edit' button
+ * that toggles an editable state. In its editable state, the fields/values are
+ * rendered as form inputs, which can be edited and then saved/persisted.
+ */
+export default class EditableSection extends Component {
+  state = {
+    data: {},
+    isEditing: false
+  }
+
+  _exists = (val) => val !== null && typeof val !== 'undefined'
+
+  _getVal = (fieldName) => this._exists(this.state.data[fieldName])
+    ? this.state.data[fieldName]
+    : this._exists(this.props.request[fieldName])
+      ? this.props.request[fieldName]
+      : ''
+
+  _onChange = (fieldName, value) => {
```

Comment:
I think these fields (e.g. credit card last four) are used for various info beyond just what they're intended for, so I hesitate to introduce form validation. I'm not as familiar with the ins and outs of formik, but I'm somewhat hesitant to refactor to use that in order to keep this component barebones.

Reply:
Formik would hook the props and event handlers to the text boxes and keep track of the field trip record state while editing, so that extra plumbing code would not be necessary. Validation is optional with Formik (and of course not required for this PR).

Reply:
Please consider changes from #328 for event and state management for this component.

Reply:
Thanks for the PR!